app-preview/src/main/java/com/mapbox/navigation/examples/preview/copilot/CopilotActivity.kt

```diff
@@ -0,0 +1,268 @@
+            NavigationOptions.Builder(this)
+                .copilotOptions(
+                    // Set shouldSendHistoryOnlyWithFeedback to true if you want to sent Copilot traces
+                    // only when an end user submits negative feedback
+                    CopilotOptions.Builder().shouldSendHistoryOnlyWithFeedback(false).build()
+                )
+                .build()
+        )
+    }
+
+    @SuppressLint("MissingPermission")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = MapboxActivityCopilotBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        copilotViewModel = ViewModelProvider(this)[CopilotViewModel::class.java]
+
+        binding.startStopNavigation.setOnClickListener {
+            when (navigationSessionState) {
+                is NavigationSessionState.Idle -> {
+                    mapboxNavigation.setNavigationRoutes(emptyList())
+                    mapboxNavigation.startTripSession()
+                }
+                is NavigationSessionState.FreeDrive -> {
+                    mapboxNavigation.stopTripSession()
+                }
+                is NavigationSessionState.ActiveGuidance -> {
+                    mapboxNavigation.setNavigationRoutes(emptyList())
+                }
+            }
+        }
+
+        binding.setRoutes.setOnClickListener {
+            buildTestRoute()
+        }
+
+        binding.feedbackPush.setOnClickListener {
+            // Call postUserFeedback every time user submits feedback
+            mapboxNavigation.postUserFeedback(
+                FeedbackEvent.POSITIONING_ISSUE,
+                "Test feedback",
+                FeedbackEvent.UI,
+                "encoded_screenshot",
+            )
+            Toast.makeText(this, "Feedback event pushed!", Toast.LENGTH_SHORT).show()
+        }
+
+        binding.searchPush.setOnClickListener {
+            // Push a SearchResultsEvent every time Search results response is retrieved
+            MapboxCopilot.push(
+                SearchResultsEvent(
+                    SearchResults("mapbox", "https://mapbox.com", null, null, "?query=test1", null)
+                )
+            )
+            // Push a SearchResultUsedEvent every time a Search result is selected
+            MapboxCopilot.push(
+                SearchResultUsedEvent(
+                    SearchResultUsed(
+                        "mapbox",
+                        "test_id",
+                        "mapbox_poi",
+                        "mapbox_address",
+                        HistoryPoint(-77.03396910343713, 38.89992797324407),
+                        null,
+                    )
+                )
+            )
+            Toast.makeText(this, "Search events pushed!", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun buildTestRoute() {
+        val origin = Point.fromLngLat(-77.0339911055176, 38.899920004207516)
```

Comment:
Route from hardcoded origin will most likely cause reroute without location simulation. Not sure if that's important in the context of this example

Reply:
Correct, it's not important in this example. CopilotActivity just shows how to integrate copilot and send events.